config/config.go

```diff
@@ -349,16 +346,10 @@ func (c *Configuration) SetEndpointAccess(e Endpoint, authType string) {
 	switch authType {
 	case "", "none":
 		git.Config.UnsetLocalKey("", key)
-
-		c.loading.Lock()
-		delete(c.gitConfig, strings.ToLower(key))
-		c.loading.Unlock()
+		c.Git.Del(key)
```

Comment:
A private func could work, as long as `SetEndpointAccess()` stays in the `config` package. I don't think it will, but I also think a better design will remove the need for it.